controlplane/kubeadm/internal/workload_cluster_conditions.go

```diff
@@ -47,8 +47,38 @@ import (
 // This operation is best effort, in the sense that in case of problems in retrieving member status, it sets
 // the condition to Unknown state without returning any error.
 func (w *Workload) UpdateEtcdConditions(ctx context.Context, controlPlane *ControlPlane) {
+	shouldRetry := func() bool {
+		// if CP is scaling up or down.
+		if ptr.Deref(controlPlane.KCP.Spec.Replicas, 0) != int32(len(controlPlane.Machines)) {
+			return true
+		}
+		// if CP machines are provisioning or deleting.
+		for _, m := range controlPlane.Machines {
+			if m.Status.NodeRef == nil {
+				return true
+			}
+			if !m.DeletionTimestamp.IsZero() {
+				return true
+			}
+		}
+		return false
+	}
+
 	if controlPlane.IsEtcdManaged() {
-		w.updateManagedEtcdConditions(ctx, controlPlane)
+		// Update etcd conditions.
+		// In case of well known temporary errors + control plane scaling up/down or rolling out, retry a few times.
+		// Note: this is required because there isn't a watch mechanism on etcd.
+		maxRetry := 3
+		for i := range maxRetry {
+			retryableError := w.updateManagedEtcdConditions(ctx, controlPlane)
+			if !retryableError {
+				break
+			}
+			// if we should retry and there is a retry left, wait a bit.
+			if shouldRetry() && i < maxRetry-1 {
+				time.Sleep(time.Duration(250*(i+1)) * time.Millisecond)
```

Comment:
This will make our Reconcile pretty slow. Worst case we're talking about 750ms sleep, correct?

We had to do a lot of work to get the *average* duration of the entire Reconcile below 1s

Reply:
We are retrying only in case KCP is not stable (scaling/up, down, provisioning/deleting) and in case of few, well know temporary errors.

Hopefully, this should not impact the average reconcile time when we have many KCP objects, and most of them are stable.

Reply:
Let's go with that for now :)